packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -147,22 +149,23 @@ export function createCloudflareEnvironmentOptions(
 				createEnvironment(name, config) {
 					return new vite.BuildEnvironment(name, config);
 				},
-				outDir: path.join('dist', name),
+				outDir: userConfig.build?.outDir ?? path.join('dist', name),
```

Comment:
This won't work for all scenarios. The `outDir` can also be set per-environment in the config e.g.

```ts
environments: {
  worker: {
    build: {
      outDir: 'something-else'
    }
  }
}
```

We should go back to how I had it before using the `configEnvironment` hook. Please see the changes that were made in this PR for how to revert them:

https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/39/files#diff-0217c39a7be6fcdb410ac8dfea2ae2998b882a1aaba3f2e83ff2831a0beee7bf

This also means we can remove the `name` and `userConfig` parameters from this function as they aren't needed.

If you want to make the code less fragmented then we could also move some of the other configuration to the `configEnvironment` hook. This is something we could also look at as part of a broader refactor later though so isn't essential now. It would give us more control over which parts of the configuration we want to be overridable.

Reply:
Ugh. Good point.

I did (a while back) look at moving the environment configuration into this hook, but at the very least you need to create an empty object in the original config hook so that Vite knows to actually run the configureEnvironment hook for each environment. That seemed like a bit of a pain.

So yes, lets just bump that bit back to the configureEnvironment and move on.

Reply:
https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/57/commits/85c74e10d4b390417d62a80242c4447fac5bc75a